backend/open_webui/routers/prompts.py

```
@@ -20,24 +20,84 @@
     if user.role == "admin":
         prompts = Prompts.get_prompts()
     else:
-        prompts = Prompts.get_prompts_by_user_id(user.id, "read")
+        # Non-admin users see public prompts + owned prompts + shared prompts
+        prompts = Prompts.get_prompts_by_user_id(user.id)
 
     return prompts
 
 
 @router.get("/list", response_model=list[PromptUserResponse])
 async def get_prompt_list(user=Depends(get_verified_user)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)